lib/pbench/test/unit/server/conftest.py

```diff
@@ -256,6 +257,34 @@ def fake_stat(file: str):
         yield
 
 
+@pytest.fixture()
+def attach_dataset(pbench_token, create_user):
+    """
+    Create test Datasets for the authorized user ("drb") and for another user
+    ("test")
+
+    Args:
+        pbench_token: create a "drb" user and authorize it
+        create_user: create a "test" user
+    """
+    Dataset(owner="drb", controller="node", name="drb", access="private").add()
+    Dataset(owner="test", controller="node", name="test", access="private").add()
+
+
+@pytest.fixture()
+def provide_metadata(monkeypatch, attach_dataset):
+    drb = Dataset.attach(controller="node", name="drb")
+    test = Dataset.attach(controller="node", name="test")
+    Metadata.create(dataset=drb, key=Metadata.SAVED, value=False)
+    Metadata.create(dataset=drb, key=Metadata.SEEN, value=True)
+    Metadata.create(dataset=drb, key=Metadata.USER, value={"contact": "me@example.com"})
+    Metadata.create(dataset=test, key=Metadata.SAVED, value=False)
+    Metadata.create(dataset=test, key=Metadata.SEEN, value=False)
+    Metadata.create(
+        dataset=test, key=Metadata.USER, value={"contact": "you@example.com"}
+    )
+
```

Comment:
Why is `monkeypatch` included in the function signature?

Reply:
Because I didn't think to remove it when I went from fully mocked Metadata objects to DB rows (necessary because of the same problem Nikhil ran into with the Dataset vs User).